expand/expand_metadata.go

```diff
@@ -123,14 +98,11 @@ func expandBoolMap(m map[string]interface{}) map[string]bool {
 	return result
 }
 
-func isKeyInMap(key string, d map[string]interface{}) bool {
-	if d == nil {
-		return false
-	}
-	for k := range d {
-		if k == key {
-			return true
+func ParseMap(aMap map[string]string, bMap map[string]interface{}) map[string]string {
```

Comment:
The name of this function is not descriptive and misleading. It is not parsing a map, it is actually removing all keys from the first parameter that are not on the second one.

Please find a more suitable name, for example `Intersection` (based on set theory https://en.wikipedia.org/wiki/Intersection_(set_theory))

Reply:
renamed them the same name that Kubernetes provider uses for the same operation